Core/HW/SasAudio.cpp

```diff
@@ -782,27 +787,38 @@ void ADSREnvelope::SetState(ADSRState state) {
 	state_ = state;
 }
 
+void ADSREnvelope::GetTypeAndRate(int &type, int &rate) {
+	switch (state_) {
+	case STATE_ATTACK: type = attackType; rate = attackRate; break;
+	case STATE_DECAY: type = decayType; rate = decayRate; break;
+	case STATE_SUSTAIN: type = sustainType; rate = sustainRate; break;
+	case STATE_RELEASE: type = releaseType; rate = releaseRate; break;
+	case STATE_KEYON:
+		height_ = 0;
```

Comment:
Seems like this should return some kind of type/rate (possibly -1 or something?)

-[Unknown]